### PR TITLE
reduce nesting in blassertsplit

### PR DESCRIPTION
**Pull request by thewilsonator**

### CODE REVIEW

**Comment by thewilsonator** [comment]
the performance of this function is `O(very bad)`, I want to refactor it a bit before making it `O(acceptable)`

**Comment by WalterBright** [comment]
I presume this is about https://issues.dlang.org/show_bug.cgi?id=19550

**Comment by thewilsonator** [comment]
Yes.

**Comment by WalterBright** [comment]
I put a link to this in 19550.

**Comment by thewilsonator** [comment]
N.B. this doesn't fix that issue, I'm just beautifying the code beforehand. I don't really understand the code other than that it does what appears to be an awful lot of nested looping.

File: src/dmd/backend/blockopt.d [hunk 1]
```diff
@@ -2213,113 +2213,119 @@ private void blassertsplit()
...
-                    bexit.BC = BCexit;
-                    bexit.Belem = e.EV.E2;
+            if (!(dctor == 0 &&   // don't split block between a dctor..ddtor pair
+                e.Eoper == OPoror && e.EV.E2.Eoper == OPcall && e.EV.E2.EV.E1.Eoper == OPvar))
+            {
+                accumDctor(e);
+                continue;
+            }
+            Symbol *f = e.EV.E2.EV.E1.EV.Vsym;
+
```
Comment (wilzbach) [inline src/dmd/backend/blockopt.d]: Trailing whitespace here:

```
Error - file 'dmd\backend\blockopt.d' contains trailing whitespace at line 2252
```

Maybe a good idea to take those 10 minutes to setup your editor to be aware of EditorConfig files.

**Pull request merged**